src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Input/InputScopeConverter.cs

```diff
@@ -109,14 +109,15 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
 
             if (null != stringSource)
             {
-                stringSource = stringSource.Trim();
+                ReadOnlySpan<char> spanSource = stringSource;
+                spanSource = spanSource.Trim();
 
-                if (-1 != stringSource.LastIndexOf('.'))
-                    stringSource = stringSource.Substring(stringSource.LastIndexOf('.')+1);
+                if (-1 != spanSource.LastIndexOf('.'))
+                    spanSource = spanSource.Slice(stringSource.LastIndexOf('.')+1);
```

Comment:
Replace stringSource.LastIndexOf('.') with spanSource.LastIndexOf('.') because at this point, if there are whitespace in stringSource, stringSource != spanSource, because of spanSource.Trim(). You could also store the result of spanSource.LastIndexOf('.') in a local variable if you want.

Reply:
Thanks, that was an oversight.  Fixed.